packages/compass-data-modeling/src/services/data-model-storage.ts

```diff
@@ -1,5 +1,27 @@
-type MongoDBDataModelDescription = unknown;
+import { z } from '@mongodb-js/compass-user-data';
+
+export const MongoDBDataModelDescriptionSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  /**
+   * Connection id associated with the data model at the moment of configuring
+   * and analyzing. No connection id means diagram was imported and not attached
+   * to a connection. Practically speaking it just means that we can't do
+   * anything that would require re-fetching data associated with the diagram
+   */
+  connectionId: z.string().nullable(),
+
+  // TODO: define rest of the schema based on arch doc / tech design
+  edits: z.array(z.unknown()).default([]),
```

Comment:
Should we add createdAt and updatedAt?

Reply:
Yes, definitely! There's an example in pipeline storage where we get this info [from the file stats](https://github.com/mongodb-js/compass/blob/16960de25cbedb24991191e1af12e0ad88a71691/packages/my-queries-storage/src/compass-pipeline-storage.ts#L24-L30), but also I guess for web we'd need to just set it on save because there are no stats there

Reply:
Noting we might not need this as we can refer to the edits in the diagram for this. Each edit will have a timestamp (first is the createdAt, last would be updatedAt).